lib/puppet/pops/model/factory.rb

```diff
@@ -431,6 +431,32 @@ def build_QueryExpression(o, expr)
   end
 
   def build_TypeAlias(o, name, type_expr)
+    if type_expr.model_class <= KeyedEntry
+      # KeyedEntry is used for the form:
+      #
+      #   type Foo = Bar { ... }
+      #
+      # The entry contains Bar => { ... } and must be transformed into:
+      #
+      #   Object[{parent => Bar, ... }]
+      #
+      parent = type_expr['key']
+      hash = type_expr['value']
+      unless parent['cased_value'] == 'Object'
+        hash['entries'] << Factory.KEY_ENTRY(Factory.QNAME('parent'), parent)
+      end
+      type_expr = Factory.QREF('Object').access([hash])
+    elsif type_expr.model_class <= LiteralHash
+      # LiteralHash is used for the form:
+      #
+      #   type Foo = { ... }
+      #
+      # The hash must be transformed into:
+      #
+      #   Object[{ ... }]
+      #
+      type_expr = Factory.QREF('Object').access([type_expr])
```

Comment:
I think the above will result in the potential issue that the `parent` key is set both by the user in the hash and from the grammar - what happens then?